src/redux/rockets/rockets.js

```diff
@@ -0,0 +1,52 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const fetchRockets = createAsyncThunk('fetchRockets', async () => {
+  const res = await fetch('https://api.spacexdata.com/v3/rockets');
+  const data = res.json();
+  return data;
+});
+
+const slice = createSlice({
+  name: 'Rockets',
+  initialState: [],
+  reducers: {
+    reserveRockets: (state, { payload }) => {
+      const newState = state.map((rocket) => {
+        if (rocket.id !== payload) {
+          return rocket;
+        }
+        return { ...rocket, reserved: true };
```

Comment:
- [ ] Kindly make sure to store only selected data in the redux store. I highly recommend you store the only needed data.
 
_**Suggestion**_
`return response.data.map((rocket) => ({
      id: rocket.rocket_id,
      name: rocket.rocket_name,
      description: rocket.description,
      reserved: false,
    }))`